recipe-server/client/control/components/MultiPicker.js

```diff
@@ -0,0 +1,234 @@
+import React, { PropTypes as pt } from 'react';
+
+export default class MultiPicker extends React.Component {
+  static propTypes = {
+    unit: pt.string.isRequired,
+    plural: pt.string.isRequired,
+    options: pt.array,
+    // from redux-form
+    value: pt.oneOfType([pt.array, pt.string]),
+    onChange: pt.func,
+  };
+
+  static getActiveSelectOptions({ options }) {
+    if (!options) {
+      return [];
+    }
+
+    const selected = [];
+    for (let i = 0; i < options.length; i++) {
+      if (options[i].selected) {
+        selected.push(options[i].value);
+      }
+    }
+
+    return selected.filter(x => x);
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      filterText: null,
+    };
+
+    this.handleApplyOption = ::this.handleApplyOption;
+    this.handleRemoveOption = ::this.handleRemoveOption;
+    this.onTextChange = ::this.onTextChange;
+    this.convertValueToObj = ::this.convertValueToObj;
+  }
+
+  onTextChange(event) {
+    const { value } = event.target;
+
+    this.setState({
+      filterText: value,
+    });
+  }
+
+  getDisplayedOptions() {
+    const {
+      options = [],
+      value,
+    } = this.props;
+
+    const selectedOptions = options.filter(option =>
+      value.indexOf(option.value) === -1);
```

Comment:
It's a bit hard for me to read some of these lines that are split like this (the functions, the variable unpacking, etc.). I think our eslint config uses 100 characters for the line length, so we should have plenty of room to keep this on one line.

Alternatively, we could split up the line like this:

```js
const selectedOptions = options.filter(
  option => value.indexOf(option.value) === -1
);
```